tools/aws/aws_comprehend.py

```diff
@@ -32,27 +41,133 @@ def main():
+
+def create_temp_transcript_file(jobName, transcript):
+    # Dump the transcript text to a text file so it can be uploaded to S3
+    tempdir = tempfile.gettempdir()
+    with open(tempdir + jobName, 'w') as tmpfile:
+        tmpfile.write(transcript)
+    return tmpfile
+
 # Serialize obj and write it to output file
 def write_json_file(obj, output_file):
     # Serialize the object
     with open(output_file, 'w') as outfile:
         json.dump(obj, outfile, default=lambda x: x.__dict__)
 
+def download_from_s3(output_uri, base_uri):
+    tarFileName = "comprehend_output.tar.gz"
+    output_key = output_uri.replace(base_uri, '')
+    s3_client = boto3.client('s3')
+
+    # get the file from s3
+    with open(tarFileName, 'wb') as f:
+        s3_client.download_fileobj('amp-test-dan', output_key, f)
+
+    # extract the contents of the .tar.gz file
+    tar = tarfile.open(tarFileName)
+    test = tar.getmembers()
+    tar.extractall()
+    tar.close()
+
+    safe_delete(tarFileName)
+
+    if len(test) > 0:
+        return test[0].name
+    else:
+        return None
+
+def run_comprehend_job(jobName, inputS3Uri, outputS3Uri, dataAccessRoleArn):
+    comprehend = boto3.client(service_name='comprehend', region_name='us-east-1')
+    print("input uri:" + inputS3Uri)
+    response = comprehend.start_entities_detection_job(
+        InputDataConfig={
+            'S3Uri': inputS3Uri,
+            "InputFormat": "ONE_DOC_PER_FILE"
+        },
+        OutputDataConfig={
+            'S3Uri': outputS3Uri
+        },
+        DataAccessRoleArn=dataAccessRoleArn,
+        JobName=jobName,
+        LanguageCode='en'
+    )
+
+    status = ''
+    output_uri = ''
+
+    while status not in ('COMPLETED','FAILED', 'STOP_REQUESTED','STOPPED'):
+        jobStatusResponse = comprehend.describe_entities_detection_job(
+            JobId=response['JobId']
+        )
+        if 'EntitiesDetectionJobProperties' in jobStatusResponse.keys():
+            status = jobStatusResponse['EntitiesDetectionJobProperties']['JobStatus']
+            output_uri = jobStatusResponse['EntitiesDetectionJobProperties']['OutputDataConfig']['S3Uri']
+            print(status)
+            time.sleep(20)
+    
+    return output_uri
+
+def read_comprehend_response(reponse_filename):
+    with open(reponse_filename) as json_file:
+        data = json.load(json_file)
+    return data
+
+def safe_delete(filename):
+    try:
+        if os.path.exists(filename):
+            print("removing: " + filename)
+            os.remove(filename)
+    except Exception as e:
+        print(e)
+        return False
+    return True
+
+def copy_to_s3(input_file, bucket, jobname):
+    s3_client = boto3.client('s3')
+    try:
+        print('before response')
+        response = s3_client.upload_file(input_file, bucket, jobname)
+        print('after response')
+    except Exception as e:
+        print(e)
+        return False
+    return True
+
 if __name__ == "__main__":
```

Comment:
No need to change now, but I wonder if we can later move copy file from/to s3 into some common facility that different AWS tools can share? In fact in the provided Galaxy tool set there is already one for copy file to s3. although I did not look into details to see how applicable it is for us.